File: src/Leetcode/Power.java
package Leetcode;

import java.util.*;

class Power
{
    public static void main(String[] args) 
    {
        try (Scanner input = new Scanner(System.in))
        {
            System.out.println("Enter a number");
            double x=input.nextFloat();
            System.out.println("Enter a power");
            int y=input.nextInt();
            double ans=getPower(x, y);
            {System.out.println(ans);}
        }
    }

    private static double getPower(double x, int n) 
    {
        return helper(x, (long) n);
    }

    private static double helper(double num, Long pow)
    {
        if (num < -10000 || num > 10000)
        {
            return 0;
        }
        if (pow==0) 
        {
            return 1;
        }
        if (pow<0)
        {
            pow=pow*-1;
            num=1/num;
        }
            if(pow%2==0){
                return helper(num*num,pow/2);
            }
            else return num*helper(num,pow-1);
    }
}

File: src/Kotlin/Leetcode/Power.kt
import java.util.*

fun main() {
    println("Enter a number")
    val number1 = Scanner(System.`in`)
    val numNumber: Double = number1.nextDouble()

    println("Enter a power")
    val number2 = Scanner(System.`in`)
    val numPower: Int = number2.nextInt()

    val ans: Double = getPower(numNumber, numPower)
    println(ans)
}

fun getPower(numNumber: Double, numPower: Int): Double {
    var pow: Long = numPower.toLong()
    var number: Double = numNumber

    if (number < -10000 || number > 10000) {
        return 0.0
    }
    if (pow == 0L) {
        return 1.0
    }
    if (pow < 0) {
        pow *= -1
        number = 1 / number
    }
    return if (pow % 2 == 0L) {
        getPower(number * number, (pow / 2).toInt())
    } else {
        number * getPower(number, (pow - 1).toInt())
    }
}
